lib/Frontend/ArgsToFrontendOptionsConverter.cpp

```diff
@@ -431,108 +432,48 @@ bool ArgsToFrontendOptionsConverter::computeFallbackModuleName() {
   return false;
 }
 
-bool ArgsToFrontendOptionsConverter::computeOutputFilenames() {
-  Optional<std::vector<std::string>> outs =
+bool ArgsToFrontendOptionsConverter::
+    computeMainAndSupplementaryOutputFilenames() {
+  Optional<std::pair<std::vector<std::string>, SupplementaryOutputPaths>> outs =
       ArgsToFrontendOutputsConverter(Args, Opts.ModuleName,
                                      Opts.InputsAndOutputs, Diags)
           .convert();
   if (!outs)
     return true;
   if (FrontendOptions::doesActionProduceOutput(Opts.RequestedAction))
-    Opts.InputsAndOutputs.setMainOutputs(*outs);
+    Opts.InputsAndOutputs.setMainAndSupplementaryOutputs(outs->first,
+                                                         outs->second);
   else
-    assert(outs->empty() &&
-           "cannot have main outputs for actions that don't produce outputs");
+    assert(outs->first.empty() && outs->second.empty() &&
+           "cannot have outputs for actions that don't produce outputs");
```

Comment:
This isn't correct; `-typecheck` produces no main outputs but may have supplementary outputs.

Reply:
Fixed, thanks for catching this.